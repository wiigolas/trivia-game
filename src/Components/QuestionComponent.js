import React, {Component} from 'react';
import { Text } from 'react-native';

export default class QuestionComponent extends Component<Props> {
  constructor() {
    super();
    this.state = {}
  }
  render(){
    return <Text>{this.props.currentQuestion}</Text>
  }
}
