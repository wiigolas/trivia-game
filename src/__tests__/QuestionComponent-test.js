import 'react-native';
import React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import sinon from 'sinon';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep'}));

configure({ adapter: new Adapter() });

import QuestionComponent from '../Components/QuestionComponent';

describe('testing the <QuestionComponent>', () => {
  let wrapper;

  beforeEach(() => {
    this.wrapper = shallow(
      <QuestionComponent currentQuestion='Question' />
    );
  });

  it('renders correctly', () => {
    expect(this.wrapper).toMatchSnapshot();
  });
  
  it('displays a question', () => {
    expect(this.wrapper.props().children).toEqual('Question');
  })
})
