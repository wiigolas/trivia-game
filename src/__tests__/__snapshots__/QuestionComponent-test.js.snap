// Jest Snapshot v1, https://goo.gl/fbAQLP
exports[`testing the <QuestionComponent> renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Question
</Text>
`;
